sunpy/database/tests/test_tables.py

```diff
@@ -204,7 +204,7 @@ def test_entries_from_fido_search_result_JSOC():
 @pytest.mark.remote_data
 def test_from_fido_search_result_block(fido_search_result):
     entry = DatabaseEntry._from_fido_search_result_block(
-        fido_search_result[0, 0][0].get_response(0)[0])
+        fido_search_result[0, 0][0].get_response(0).blocks[0])
```

Comment:
What caused this change and the `_data` change?

Reply:
I see it was the new ABC. 

Does this mean a user has to worry about this?

Reply:
It's the fact that slicing the `BaseQueryResponse` object should always return an instance of itself. In this current implementation I haven't standardised any underlying storage, so the way that each subclass of the base exposes a way to slice down to the underlying data is not specified. Perhaps I should add this to the class.

Reply:
How much effort would that be?

Reply:
Not much, can be done.

Reply:
It is done.